### PR TITLE
Handle repo warning messages from zypper

### PR DESCRIPTION
**Pull request by mloviska**
- problem:[wrongly parsed zypper table](https://openqa.opensuse.org/tests/1206590#step/orphaned_packages_check/11)
- Verification runs: 
   * [sle-15-SP2-JeOS-for-kvm-and-xen-x86_64](https://openqa.suse.de/tests/4013002#step/orphaned_packages_check/1)
   * [opensuse-Tumbleweed-JeOS-for-kvm-and-xen-x86_64](https://openqa.opensuse.org/tests/1208268#step/orphaned_packages_check/1)
   * [opensuse-15.2-JeOS-for-kvm-and-xen-x86_64](https://openqa.opensuse.org/tests/1208351#step/orphaned_packages_check/1)

### CODE REVIEW

**Comment by Vogtinator** [comment]
It might actually be as simple as just adding `--no-refresh` to avoid those messages

**Comment by mloviska** [comment]
> It might actually be as simple as just adding `--no-refresh` to avoid those messages

I haven't thought about this option. Shouldn't the test ensure that the repositories are refreshed when it comes to orphan detection?
For instance, QAM doesn't use any mirrored or build repositories.

**Comment by Vogtinator** [comment]
> > It might actually be as simple as just adding `--no-refresh` to avoid those messages
> 
> I haven't thought about this option. Shouldn't the test ensure that the repositories are refreshed when it comes to orphan detection?
> For instance, QAM doesn't use any mirrored or build repositories.

Yes, but we already have `console/zypper_ref` running earlier in the test (both on SLE and openSUSE), so unless there is some update released right in between, this should work fine.

**Comment by mloviska** [comment]
> > > It might actually be as simple as just adding `--no-refresh` to avoid those messages
> > 
> > 
> > I haven't thought about this option. Shouldn't the test ensure that the repositories are refreshed when it comes to orphan detection?
> > For instance, QAM doesn't use any mirrored or build repositories.
> 
> Yes, but we already have `console/zypper_ref` running earlier in the test (both on SLE and openSUSE), so unless there is some update released right in between, this should work fine.

`console/zypper_ref` is one of the test cases used almost everywhere. If we are fine creating this dependency, let's do it the easier way. 
I was happy about removing the fragile `awk` part. However, I admit that regex is not a bulletproof solution.

**Comment by Vogtinator** [comment]
Yeah, it would be great to have `zypper --xmlout packages` work, but it's not really implemented properly...

**Comment by mloviska** [comment]
> Yeah, it would be great to have `zypper --xmlout packages` work, but it's not really implemented properly...

or something like `lsblk --json`.

File: tests/console/orphaned_packages_check.pm [hunk 1]
```diff
@@ -65,9 +66,9 @@ sub run {
     # Orphans are also expected on JeOS without SDK module (jeos-firstboot, jeos-license and live-langset-data)
     # Save the orphaned packages list to one log file and upload the log, so QA can use this log to report bug
     my @orphans = split('\n',
-        script_output q[zypper --quiet packages --orphaned | 
+        script_output q[zypper --no-refresh --quiet packages --orphaned | 
                             grep -v "\(release-DVD\|release-dvd\|openSUSE-release\|skelcd\)" |
-                            tee -a /tmp/orphaned.log | awk -F "|" 'NR>2 {print $3}'], proceed_on_failure => 1);
+                            tee -a /tmp/orphaned.log  | awk -F "|" 'NR>2 {print $3}'], proceed_on_failure => 1);
```
Comment (Vogtinator) [inline tests/console/orphaned_packages_check.pm]: Added whitespace error?

Comment (mloviska) [reply tests/console/orphaned_packages_check.pm]: happened during copy & paste. Thanks!

**Comment by pdostal** [comment]
QAM Public Cloud verification run: [SLES 12 SP4 @ GCE](http://pdostal-server.suse.cz/tests/7784) [SLES 15 SP1 @ EC2](http://pdostal-server.suse.cz/tests/7780)

**Comment by mloviska** [comment]
> It might actually be as simple as just adding `--no-refresh` to avoid those messages

```
# cat > /tmp/scriptc3Y7I.sh << 'EOT_c3Y7I'; echo c3Y7I-$?-
> zypper --no-refresh --quiet packages --orphaned | 
>                             grep -v "\(release-DVD\|release-dvd\|openSUSE-release\|skelcd\)" |
>                             tee -a /tmp/orphaned.log  | awk -F "|" 'NR>2 {print $3}'
> EOT_c3Y7I
c3Y7I-0-
```

cat /tmp/orphaned.log
```
Warning: Repository 'Main Update Repository' appears to be outdated. Consider using a different mirror or server.
Warning: Repository 'Update Repository (Non-Oss)' appears to be outdated. Consider using a different mirror or server.
S | Repository | Name                           | Version          | Arch  
--+------------+--------------------------------+------------------+-------
```

Didn't help

**Comment by Vogtinator** [comment]
> > It might actually be as simple as just adding `--no-refresh` to avoid those messages
> ...
> cat /tmp/orphaned.log
> 
> ```
> Warning: Repository 'Main Update Repository' appears to be outdated. Consider using a different mirror or server.
> Warning: Repository 'Update Repository (Non-Oss)' appears to be outdated. Consider using a different mirror or server.
> S | Repository | Name                           | Version          | Arch  
> --+------------+--------------------------------+------------------+-------
> ```
> 
> Didn't help

Meh - would've been too easy... I'll double-check whether some other option works around this.

**Comment by Vogtinator** [comment]
What about simply `zypper -q packages --orphaned | grep -v ^Warning`?

**Comment by pdostal** [comment]
LGTM from QAM Public Cloud side as well. See [12-SP4@GCE](http://pdostal-server.suse.cz/tests/7802) & [15-SP1@AWS](http://pdostal-server.suse.cz/tests/7803) fresh runs.

**Comment by mloviska** [comment]
gr8, thanks for support @Vogtinator @pdostal

**Pull request merged**